### PR TITLE
feat: Authentication

### PR DESCRIPTION
**Pull request by Tanjum786**
###  This PR contains authentication

- user can 
  - login
  - signup to the page
  - log out 
- added require auth to the private routes 
  - Cart 
  - wishlist 

[live link](https://e-galaxy-store-ecom-ganbxlzti-tanjum786.vercel.app/)

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1 +1 @@
-REACT_APP_JWT_SECRET=<your-jwt-secret>
\ No newline at end of file
+REACT_APP_JWT_SECRET=ecommerceapp
```
Comment (omjadhav85) [inline .env:1]: You should ideally include .env in gitignore file and put all your environment variables on the deployed server (Vercel in your case)

File: src/components/Navbar.jsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                  <i className="icon_nav fas fa-shopping-cart"></i>
+                </Link>
+                <p className="badge_cart font_small dis_flex padding_small">
+                  {cart.length}
+                </p>
+              </div>
+            </li>
+            <li className="list-item-inline">
+              <div className="user-container dis_flex">
+                <Link to="/add usre">
```
Comment (omjadhav85) [inline src/components/Navbar.jsx:91]: Seems like an incorrect link path

File: src/Context/context/Authcontext.jsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  const location = useLocation();
+  const [authState, Authdispatch] = useReducer(authReducer, {
+    email: "",
+    firstName: "",
+    lastName: "",
+    password: "",
+    confirPassword: "",
+    isSubmit: false,
+  });
+
```
Comment (omjadhav85) [inline src/Context/context/Authcontext.jsx:24]: I don't think these details are needed to be stores in context as they wont be needed everytime and you are already storing the user details in local storage. You can define a local state variable for this in SignUp component itself.

**Review by omjadhav85 — APPROVED** [review]
Great work! Features working as expected. Provided some sughestions which can be implemeted. Also, just FYI the SignUp component is incorrectly named, you might want to rename it to avoid issues later. 

Other than that, looks good to merge!

**Review by NamrataRaikwar2002 — APPROVED** [review]

File: src/Context/context/Authcontext.jsx (hunk 3 above)
Comment (Tanjum786) [reply src/Context/context/Authcontext.jsx:24]: yeah, i will change it.thank you for pointing out

**Pull request merged**